tokio-test/src/io.rs

```diff
@@ -453,6 +453,16 @@ impl AsyncWrite for Mock {
     }
 }
 
+/// Ensures that Mock isn't dropped with data "inside".
+impl Drop for Mock {
+    fn drop(&mut self) {
+        self.inner.actions.iter().for_each(|a| match a {
+            Action::Read(data) if !data.is_empty() => panic!("There is still data left to read."),
+            Action::Write(data) if !data.is_empty() => panic!("There is still data left to write."),
```

Comment:
nit, take it or leave it: what do you think of expressing this as
```suggestion
            Action::Read(data) => assert!(data.is_empty(), "There is still data left to read."),
            Action::Write(data) => assert!(data.is_empty(), "There is still data left to write."),
```
it's pretty much equivalent, but an "assertion failed" message might make it clearer to the reader that the test was _supposed_ to fail.